packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenLock.sol

```diff
@@ -0,0 +1,492 @@
+        emit SetRequesterAuthorizerWithManager(
+            _chainId,
+            _requesterAuthorizerWithManager,
+            msg.sender
+        );
+    }
+
+    /// @notice Called by the blockWithdrawDestination setter to set the
+    /// withdraw destination of blocked requester tokens
+    /// @param _destination The destination to which the blocked requester tokens will be withdrawn
+    function setBlockWithdrawDestination(address _destination)
+        external
+        override
+    {
+        require(
+            hasBlockWithdrawDestinationSetterRoleOrIsManager(msg.sender),
+            ERROR_NOT_BLOCK_WITHDRAW_DESTINATION_SETTER
+        );
+        require(_destination != address(0), ERROR_ZERO_ADDRESS);
+        blockWithdrawDestination = _destination;
+        emit SetBlockWithdrawDestination(_destination, msg.sender);
+    }
+
+    /// @notice Locks API3 Tokens to gain access to Airnodes.
+    /// @dev chainId-airnode-endpoint-requester pair gets authorized as long as there is
+    /// at least one token lock for given pair.
+    /// @notice The amount to be locked is determined by the fee set in the AirnodeFeeRegistry Contract
+    /// @param _chainId The id of the chain
+    /// @param _airnode The airnode address
+    /// @param _endpointId The endpointId
+    /// @param _requesterAddress The address of the requester for which tokens are being locked
+    function lock(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress
+    )
+        external
+        override
+        isOptedIn(_airnode)
+        isNotBlocked(_airnode, _requesterAddress)
+    {
+        require(_chainId != 0, ERROR_ZERO_CHAINID);
+        require(_airnode != address(0), ERROR_ZERO_ADDRESS);
+        require(_requesterAddress != address(0), ERROR_ZERO_ADDRESS);
+        TokenLocks storage tokenLock = tokenLocks[_chainId][_airnode][
+            _endpointId
+        ][_requesterAddress];
+        require(
+            tokenLock.lockerToLockAmount[msg.sender] == 0,
+            ERROR_ALREADY_LOCKED
+        );
+
+        uint256 lockAmount = getLockAmount(_chainId, _airnode, _endpointId);
+
+        require(
+            IERC20(api3Token).balanceOf(msg.sender) >= lockAmount,
+            ERROR_INSUFFICIENT_AMOUNT
+        );
+
+        tokenLock.lockerToLockAmount[msg.sender] = lockAmount;
+        tokenLock.whitelistCount++;
+
+        if (tokenLock.whitelistCount == 1) {
+            IRequesterAuthorizerWithManager(
+                chainIdToRequesterAuthorizerWithManager[_chainId]
+            ).setIndefiniteWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress,
+                    true
+                );
+        }
+
+        assert(
+            IERC20(api3Token).transferFrom(
+                msg.sender,
+                address(this),
+                lockAmount
+            )
+        );
```

Comment:
We can `require()` with an error message here assuming one can use another ERC20 implementation instead of API3 where `transferFrom` returns `false` instead of reverting

Reply:
api3Token is immutable so its not possible that some other token is being used unless the deployer put in the wrong address

Reply:
"api3Token" may not be API3 but a different ERC20